ScrapeRepos.ts

```diff
@@ -0,0 +1,356 @@
+ * @returns {string} the summary by ChatGPT of the repo
+ */
+async function openAIRequestTurbo(readME: String, OPENAI_API_KEY: string) {
+  const question =
+    "The following text describes a prorgamming project that is current in development. Explain to me what the project is trying to archieve without telling me" +
+    "how they are doing so. Please use arround 50 words and don't get too technical" +
+    readME;
+  const response = await fetch("https://api.openai.com/v1/chat/completions", {
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: "Bearer " + OPENAI_API_KEY
+    },
+    method: "POST",
+    body: JSON.stringify({
+      model: "gpt-3.5-turbo",
+      messages: [
+        {
+          role: "system",
+          content:
+            "You are a Computer Science Teacher to his students without tehcnical knowledge",
+        },
+        {
+          role: "user",
+          content: question,
+        },
+      ],
+    }),
+  });
+  const data = await response.json();
+  const content = data.choices[0].message.content;
+  return content
+}
+
+
+/** Method to scrape the star history from a repo
+ * @param {string} owner Owner of the repository
+ * @param {string} repo Repository name
+ * @param {string} authToken AuthToken for GitHub API
+ * @returns {[date: 'DATE', count: 'COUNT']}: { date: '2023-05-10', count: 1549 } 
+ * => on error empty array and error message
+ */
+async function fetchRepositoryStars(owner: string, repo: string, authToken: string) {
+  const query = `
+    query ($owner: String!, $repo: String!, $cursor: String) {
+      repository(owner: $owner, name: $repo) {
+        stargazers(first: 100, orderBy: { field: STARRED_AT, direction: DESC }, after: $cursor) {
+          totalCount
+          edges {
+            starredAt
+            node {
+              login
+            }
+          }
+          pageInfo {
+            endCursor
+            hasNextPage
+          }
+        }
+      }
+    }
+  `;
+
+  const headers = {
+    'Authorization': authToken
+  };
+
+  let hasNextPage = true;
+  let endCursor: string | null = null;
+
+  let starHistoryList: any[] = []
+
+  try {
+    while (hasNextPage) {
+      const variables: any = {
+        owner,
+        repo,
+        cursor: endCursor
+      };
+
+      const response = await axios.post(
+        'https://api.github.com/graphql',
+        { query, variables },
+        { headers }
+      );
+      const starsData = response.data.data.repository.stargazers.edges;
+      starHistoryList.push(starsData)
+
+      const pageInfo = response.data.data.repository.stargazers.pageInfo;
+      hasNextPage = pageInfo.hasNextPage;
+      endCursor = pageInfo.endCursor || null;
+
+      variables.cursor = endCursor;
+    }
+  } catch (error) {
+    console.error('Error:', error);
+    return []
+  }
+
+  // get the real star history
+  const mergedArray = [].concat(...starHistoryList);
+  const result = await starHistory(mergedArray);
```

Comment:
Please directly return the result of function. You should not create a new variable.